### PR TITLE
v1.1.0

### PR DESCRIPTION
**Pull request by max-degterev**

### CODE REVIEW

**Review by peterhass — APPROVED** [review]

**Comment by max-degterev** [comment]
Known issues:
- [x]  Rebuilding doesn't remove deleted modules from output, need diffing

File: fixtures/app/app.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import translations from './app.translations';
+import dictionary from '../dictionaries/en-US';
+
+console.log(translations, dictionary);
```
Comment (peterhass) [inline fixtures/app/app.js]: Write something to the DOM. I usually do `npm start` without looking at the code first and thought the preview was dead.

'Check console' would have been sufficient

File: lib/schema.json [hunk 2]
```diff
@@ -2,18 +2,25 @@
...
       "minLength": 1,
       "pattern": "\\.json$"
     },
     "dictionaryPattern": {
+      "description": "Absolute or relative path to the dictionary files with [locale_code] in file name.",
       "type": "string",
       "minLength": 1,
       "pattern": "\\[locale_code\\].*\\.json$"
+    },
+    "emitFile": {
```
Comment (peterhass) [inline lib/schema.json:21]: Not sure about the setting name. To `emit` a file in webpack usually means to produce something like an asset in the build folder. This setting actually changes a **source file** 

Don't have a good suggestion. Maybe `patchDictionaries`?

File: lib/sync.js [hunk 3]
```diff
@@ -85,39 +93,38 @@ const update = (options) => {
...
-  debug('found %d dictionaries, attempting to update modules', Object.keys(files).length);
+  debug('found %d module ids, attempting to update module files', Object.keys(moduleUpdates).length);
 
-  Object.keys(files).forEach((id) => {
-    const filePath = path.resolve(reverseModuleId(id, options.keysRoot, options.moduleEnding));
-    const content = JSON.stringify(files[id], null, 2);
-    debug('saved %s module to %s', id, filePath);
+  Object.keys(moduleUpdates).forEach((id) => {
+    const filePath = moduleFiles[id];
+    if (!filePath) return debug('module file for %s doesn\'t exist', id);
```
Comment (peterhass) [inline lib/sync.js]:
```suggestion
    if (!filePath) { 
      debug('module file for %s doesn\'t exist', id);
      return;
    }
```

File: lib/schema.json (hunk 2 above)
Comment (max-degterev) [reply lib/schema.json:21]: -> `updateDictionaries`?
it is an asset in a way. not sure

File: lib/sync.js (hunk 3 above)
Comment (max-degterev) [reply lib/sync.js]: i will pad with spaces a bit, i see no reason to split into multiple lines

Comment (peterhass) [reply lib/sync.js]: My point is to not return values that aren't used

Comment (max-degterev) [reply lib/sync.js]: i agree in general, but within a forEach it doesn't matter

File: lib/schema.json (hunk 2 above)
Comment (max-degterev) [reply lib/schema.json:21]: will keep it for now, it's an asset, existing in manifest.json and so on

**Review by atomiomi — APPROVED** [review]

File: lib/plugin.js [hunk 4]
```diff
@@ -1,147 +1,127 @@
...
-      // There is probably a better way to do this, but there is almost no documentation 🤷‍♂️
-      compilation.hooks.optimizeChunkAssets.tapAsync(PLUGIN_NAME, (chunks, callback) => {
-        const replacements = Array.from(this.seeds.keys());
-        if (!replacements.length) return callback();
-
-        debug('attempting to find patterns to replace');
+      compilation.hooks.finishModules.tapPromise(PLUGIN_NAME, async(modules) => {
+        const builtModules = modules.reduce((acc, webpackModule) => {
+          if (!webpackModule.resource || !webpackModule.resource.endsWith(moduleEnding)) return acc;
+          acc.push(getModuleId(keysRoot, moduleEnding, webpackModule.resource));
```
Comment (peterhass) [inline lib/plugin.js]: reverse if so that you only have one `return acc`

Comment (peterhass) [reply lib/plugin.js]: and maybe extract into a `util` ?

Comment (max-degterev) [reply lib/plugin.js]: inverse logic is no good, will clean up. not so hot on moving this into utils, we already have `collectModules` which is also basically a 1 reduce function... will think about it

**Pull request merged**